src/main/java/de/unistuttgart/overworldbackend/data/Course.java

```diff
@@ -23,14 +24,26 @@ public class Course {
   @NotNull
   String courseName;
 
+  @Pattern(regexp = "^(WS|SS)-[0-9][0-9]$")
```

Comment:
```suggestion
  @Pattern(regexp = "^(WS|SS)-[0-9][0-9]+$")
```

Reply:
why the `+`? Can you give an example what your regex handles what mine does not?

Reply:
What I just noticed: We can add a unique constraint now combining both the lecture name and semester.
This avoids errors as having two lectures with the same name in the same semester is probably not a good idea, and most likely done by mistake.